gobblin-core/src/main/java/gobblin/metrics/GobblinMetrics.java

```diff
@@ -548,6 +557,155 @@ private void buildKafkaMetricReporter(Properties properties) {
+            .withConnection(hostname, port)
+            .withMetricContextName(this.metricContext.getName()) //contains the current job id
+            .build(properties);
+      }
+      catch (IOException e) {
+        LOGGER.error("Failed to create Graphite metrics reporter. Will not report metrics to Graphite.", e);
+      }
+    }
+    else {
+      LOGGER.info("Not reporting metrics to Graphite");
+    }
+        
+    if (eventsEnabled) {
+      boolean emitValueAsKey =
+          Utils.getPropAsBoolean(properties, ConfigurationKeys.METRICS_REPORTING_GRAPHITE_EVENTS_VALUE_AS_KEY,
+              ConfigurationKeys.DEFAULT_METRICS_REPORTING_GRAPHITE_EVENTS_VALUE_AS_KEY);
+      String eventsPortProp = properties.getProperty(ConfigurationKeys.METRICS_REPORTING_GRAPHITE_EVENTS_PORT);
+      int eventsPort =
+          (eventsPortProp == null) ? (metricsEnabled ? port : 
+            Integer.parseInt(ConfigurationKeys.METRICS_REPORTING_GRAPHITE_PORT)) : Integer.parseInt(eventsPortProp);
+      try {
+        GraphiteEventReporter eventReporter =
+            GraphiteEventReporter.Factory.forContext(this.metricContext)
+              .withConnectionType(connectionType)
+              .withConnection(hostname, eventsPort)
+              .withEmitValueAsKey(emitValueAsKey)
+              .build();
+        this.scheduledReporters.add(this.codahaleReportersCloser.register(eventReporter));
+      }
+      catch (IOException e) {
+        LOGGER.error("Failed to create Graphite event reporter. Will not report events to Graphite.", e);
+      }
+    }
+    else  {
+      LOGGER.info("Not reporting events to Graphite");
+    }
+  }
+
+  private void buildInfluxDBMetricReporter(Properties properties) {
+    boolean metricsEnabled =
+        Utils.getPropAsBoolean(properties, ConfigurationKeys.METRICS_REPORTING_INFLUXDB_METRICS_ENABLED_KEY,
+            ConfigurationKeys.DEFAULT_METRICS_REPORTING_INFLUXDB_METRICS_ENABLED);
+    boolean eventsEnabled =
+        Utils.getPropAsBoolean(properties, ConfigurationKeys.METRICS_REPORTING_INFLUXDB_EVENTS_ENABLED_KEY,
+            ConfigurationKeys.DEFAULT_METRICS_REPORTING_INFLUXDB_EVENTS_ENABLED);
+    if (!metricsEnabled && !eventsEnabled) {
+      LOGGER.info("Not reporting metrics and events to InfluxDB");
+      return;
+    }
+    
+    try {
+      Preconditions.checkArgument(properties.containsKey(ConfigurationKeys.METRICS_REPORTING_INFLUXDB_DATABASE),
+          "InfluxDB database name is missing.");
+    } catch (IllegalArgumentException exception) {
+      LOGGER.error("Not reporting to InfluxDB due to missing InfluxDB configuration(s).", exception);
+      return;
+    }
+    
+    String url = properties.getProperty(ConfigurationKeys.METRICS_REPORTING_INFLUXDB_URL);
+    String username = properties.getProperty(ConfigurationKeys.METRICS_REPORTING_INFLUXDB_USER);
+    String password = properties.getProperty(ConfigurationKeys.METRICS_REPORTING_INFLUXDB_PASSWORD);
```

Comment:
Not necessary, but would it be good to allow encryption of this password? (There is a password manager module in Gobblin).